R/importRecords.R

```diff
@@ -208,17 +210,23 @@ importRecords.redcapApiConnection <- function(rcon,
     which(names(data) %in% 
             c("redcap_survey_identifier",
               paste0(unique(MetaData$form_name), "_timestamp")))
-  if (length(w.remove)) data <- data[-w.remove]
+  if (length(w.remove) > 0) data <- data[-w.remove]
+  
+  mchoices <- which(vapply(data, inherits, logical(1), 'mChoice'))
+  if(length(mchoices) > 0)
+  {
+    stop("The variable(s) ", 
+         paste0(names(data)[mchoices], collapse=", "), 
+         " are mChoice formatted and cannot be imported.")
+  }
   
   # Validate field names
   unrecognized_names <- !(names(data) %in% c(with_complete_fields, REDCAP_SYSTEM_FIELDS))
-
   if (any(unrecognized_names))
   {
-    message("The variable(s) ", 
+    stop("The variable(s) ", 
```

Comment:
Same as previous comment. Could this be `coll$push` instead of `stop`. That would allow it to continue with validation and report all of the failed validations at line 283.